### PR TITLE
Add ectoplasm color scheme (Method A)

### PR DESCRIPTION
**Pull request by mreishus**
One part of #47036
Method A is defined in #47103. There's a "Method B" copy of the PR here: #47075.

#### Changes proposed in this Pull Request

* Add ectoplasm color scheme

#### Testing instructions

* Visit /me/account
* Choose ectoplasm and save

To see the wp-admin color scheme, create an atomic site, then visit /wp-admin/profile.php, choose color scheme and save.

Calypso:

Wp-Admin:

#### Known Issues

* Big: All lightness values of both primary and accent colors, -10 through -90, are set to the same color.  How do we generate these while maintaining a seamless experience between wp-admin and calypso?  [Here's how color-studio generates them](https://github.com/Automattic/color-studio/blob/master/data/color-definitions.js).
* Primary and accent colors are the same.  Does this cause an issue?

### CODE REVIEW

**Comment by mreishus** [comment]
Listing issues that arise from setting all primary-10 - primary-90 and accent-10 - accent-90 colors to wp-admin highlight color.  (Starting with one and will edit others in)

Hover tab bar - Unreadable text

**Comment by mreishus** [comment]
So, having all -10 -90 assigned to the same color was causing some issues as illustrated above.  I used color-studio to generate a range of colors and it fixes the issue.

Since we set the sidebar colors exactly, we can make sure that those match wp-admin exactly and don't use the "palette" colors.

in `data/color-definitions.js`
```
    {
      name: 'Ectoplasm Green',
      default: 50,
      specs: {
        hue_start: 70,
        hue_end: 72,
        hue_curve: 'easeOutSine',
        sat_steps: [
          8, 27, 56, 80, 100, 100, 100, 100, 100, 100, 100, 100
        ],
        lum_steps: [
          96, 96, 95, 94, 87, 75.1, 61.6, 49, 40.5, 31, 20, 11
        ]
      }
    }
```

**Review by sixhours — CHANGES_REQUESTED** [review]
Took this for a quick spin! I noticed the outline around the help button could be tweaked to use a variation of the green:

Screen Shot 2020-11-06 at 9 03 06 AM

What do you think about adding a darker purple for the top bar border color to give it a little separation from the side nav?

Screen Shot 2020-11-06 at 9 00 57 AM

Screen Shot 2020-11-06 at 9 21 01 AM

**Review by cpapazoglou — CHANGES_REQUESTED** [review]
Thanks @mreishus for wrangling this!
Generally it works as expected. I navigated around and checked the components page! Let's check if we can easily address the following:

Menu Heading should be the same as links or at least vivid enough

**Comment by cpapazoglou** [comment]
> Took this for a quick spin! I noticed the outline around the help button could be tweaked to use a variation of the green:
> 
> Screen Shot 2020-11-06 at 9 03 06 AM
> 
> What do you think about adding a darker purple for the top bar border color to give it a little separation from the side nav?
> 
> Screen Shot 2020-11-06 at 9 00 57 AM
> 
> Screen Shot 2020-11-06 at 9 21 01 AM

@sixhours thanks for chiming in! As it seems all wp-admin color schemes make no distinction between the masterbar and the sidebar in terms of colour or bordering. I agree it would be best, but I am not sure that we should start making changes to already wp-admin defined colour schemes. This would increase maintainability of schemes and make the scope of porting the colour schemes even larger. @davemart-in , @lcollette what is your take on this?

File: packages/calypso-color-schemes/src/shared/color-schemes/_ectoplasm.scss [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+:root {
+    --theme-text-color: #ffffff; /* Direct from wp-admin */
+    --theme-text-color-rgb: 255, 255, 255; /* Manual conversion */
+    --theme-base-color: #523f6d; /* Direct from wp-admin */
+    --theme-base-color-rgb: 82, 63, 109; /* Manually computed https://www.rapidtables.com/convert/color/hex-to-rgb.html */
+    --theme-submenu-background-color: #413256; /* darken( $base-color, 7% ), computed: http://scg.ar-ch.org/
+    --theme-icon-color: #ece6f6; /* Direct from wp-admin */
+    --theme-highlight-color: #a3b745; /* Direct from wp-admin */
+    --theme-highlight-color-rgb: 163, 183, 69; /* Manually computed https://www.rapidtables.com/convert/color/hex-to-rgb.html */
+    --theme-notification-color: #d46f15; /* Direct from wp-admin */
```
[the remark this answers is not in the document]
Comment (mreishus) [reply packages/calypso-color-schemes/src/shared/color-schemes/_ectoplasm.scss:45]: I'm not sure I understand, these are specific for each theme.  I define them at the top and use them later in the theme file.
One exception is the text color, all of the wp-admin themes use the same text color (so far).

**Comment by sixhours** [comment]
The /wp-admin version of Ectoplasm shows a border for me:

Screen Shot 2020-11-06 at 10 36 29 AM

File: packages/calypso-color-schemes/src/shared/color-schemes/_ectoplasm.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (mreishus) [reply packages/calypso-color-schemes/src/shared/color-schemes/_ectoplasm.scss:45]: We talked about this a little more in IM and discussed a solution :)

**Comment by cpapazoglou** [comment]
> The /wp-admin version of Ectoplasm shows a border for me:
> 
> Screen Shot 2020-11-06 at 10 36 29 AM

Interesting, the core version doesn't have one... It is a wpcom hack. 


That said, I am not sure how we should proceed... `wpcom-admin-bar.css` contains a dozen of overrides.

**Comment by mreishus** [comment]
### **Variable scope change**
Done.
### **Notification circle color in masterbar:**
Done.
### **Masterbar bottom border color:**
As noted above, it seems like this looks different depending on if you're on atomic or not. I'm okay adding a bottom border since there is already support in it in calypso, and I only have to change one line.  **I've made the change here.**  I guess it ultimately depends on which version of WP we want to be consistent with.
### FAB Outer ring color
It is already based on the green highlight color, but can be hard to tell: 

It doesn't seem too out of line with an existing theme in calypso, like sunset:

This one I'm hesitant to change because I'd have to make `--color-primary-dark` lighter, which might affect readability of some text (like the CreditCard code listing change a few posts up)


### **Sidebar top level menu text brightness**
I can do this, (And I have), but I'm wondering if it's the right thing to do.

In calypso, it's normal for the top level menus to have less contrast than the inner menus.  Examples:



In wp-admin, I see the outer menus brighter

 
So we have three options for the ports
A (Dimmer outer, conforms to calypso feeling)
2020-11-06_11-14-bright-inner

B (Dimmer inner, conforms to wp-admin feeling)
2020-11-06_11-12-bright-outer

C (Both bright, compromise)
2020-11-06_11-11-same-bright

What should we go with?  I have "C" pushed right now.

**Comment by mreishus** [comment]
:man_facepalming: I just realized that these don't work when the nav unification flag is on, because [these overrides](https://github.com/Automattic/wp-calypso/blob/6d4d5f8fa0094aca366a4ed221985a201dbc822e/client/my-sites/sidebar-unified/style.scss#L4-L28) take precedence.  I made a crude fix that re-specifies all the theme variables inside a `.is-nav-unification { }` selector, but will think about a better fix on Monday.

**Comment by cpapazoglou** [comment]
> 🤦‍♂️ I just realized that these don't work when the nav unification flag is on, because [these overrides](https://github.com/Automattic/wp-calypso/blob/6d4d5f8fa0094aca366a4ed221985a201dbc822e/client/my-sites/sidebar-unified/style.scss#L4-L28) take precedence. I made a crude fix that re-specifies all the theme variables inside a `.is-nav-unification { }` selector, but will think about a better fix on Monday.

Changes in `sidebar-unified/style.scss` will ultimately reside in the default theme `_default.scss` and they will be overridden by each colour-scheme. Till then, I pushed a commit, making your idea DRY. It should be enough.

**Review by cpapazoglou — APPROVED** [review]
Thanks for pushing the changes @mreishus!

LGTM!

**Pull request merged**